src/dbus_connection.rs

```diff
@@ -0,0 +1,355 @@
+                    method_call_closure(
+                        conn,
+                        sender,
+                        object_path,
+                        interface_name,
+                        method_name,
+                        parameters,
+                        invocation,
+                    );
+                    None
+                })
+                .to_glib_none()
+                .0,
+                glib::Closure::new(move |args| {
+                    let mut iter = args.iter();
+                    let conn = iter
+                        .next()
+                        .unwrap()
+                        .get::<DBusConnection>()
+                        .unwrap()
+                        .unwrap();
+                    let sender = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let object_path = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let interface_name = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let property_name = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let result = get_property_closure(
+                        conn,
+                        sender,
+                        object_path,
+                        interface_name,
+                        property_name,
+                    );
+                    Some(result.to_value())
+                })
+                .to_glib_none()
+                .0,
+                glib::Closure::new(move |args| {
+                    let mut iter = args.iter();
+                    let conn = iter
+                        .next()
+                        .unwrap()
+                        .get::<DBusConnection>()
+                        .unwrap()
+                        .unwrap();
+                    let sender = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let object_path = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let interface_name = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let property_name = iter.next().unwrap().get::<&str>().unwrap().unwrap();
+                    let value = iter
+                        .next()
+                        .unwrap()
+                        .get::<glib::Variant>()
+                        .unwrap()
+                        .unwrap();
+                    let result = set_property_closure(
+                        conn,
+                        sender,
+                        object_path,
+                        interface_name,
+                        property_name,
+                        value,
+                    );
+                    Some(result.to_value())
+                })
+                .to_glib_none()
+                .0,
+                &mut error,
+            );
+            if error.is_null() {
+                Ok(RegistrationId(NonZeroU32::new_unchecked(id)))
```

Comment:
Maybe safer to use `new(id).expect' ?

Reply:
This is what is used in `glib`.

I don't think we really need to test gio's function contracts on every call :)

Reply:
`glib` checked it before use `new_unchecked`

Reply:
Yeah either add an assertion before and `new_unchecked()`, or `new()` and `expect()`. glib didn't do the latter because in older Rust you don't get too useful source file / line number with `expect()`.